torch/_dynamo/variables/ctx_manager.py

```diff
@@ -958,11 +958,22 @@ def call_method(
         from ..utils import proxy_args_kwargs
         from .builder import wrap_fx_proxy_cls
 
-        if name in ("wait_stream", "synchronize", "wait_event"):
+        if name in ("wait_stream", "synchronize"):
             tx.output.create_proxy(
                 "call_method", name, *proxy_args_kwargs([self] + args, kwargs)
             )
             return variables.ConstantVariable(None)
+        elif name == "wait_event":
+            # NOTE: Unlike CUDA streams, currently there is no way to look up a previously-created CUDA event by id
+            # (i.e. CUDA events are not cached in a pool), therefore there is no way to create a proxy for it and
+            # reconstruct it within the graph. As a result, Dynamo graph does not support calling `.wait_event()`
```

Comment:
I might be missing something, but is the correctness still guaranteed? (e.g. all event has been waited prior to this point)

Reply:
I believe events are not waited even before this PR, so functionally it still behaves the same.

Reply:
also prior to this PR, if the CUDA event was created outside of the graph, we are issuing `.wait_event(None)` which throws error. With changes in this PR, we will not issue that `.wait_event(None)` line, thus eliminating the error.